_posts/reflect-92.md

```diff
@@ -0,0 +1,202 @@
+---
+title: Code-First API Documentation with Swagger
+categories: [spring-boot]
+date: 
+modified: 
+author: petros
+excerpt: "What's the 'Code-First' Approach? And how do we go about it with Springdoc and Spring Boot? This guide explains and showcases this approach."
+image:
+  auto: 0035-switchboard
+tags: ["spring-boot", "code-first-approach", "swagger"]
+---
+
+When following a Code-First approach, we first start with the development, and then we might generate the specification, which then becomes the documentation.
+
+Let's see, what are the benefits of using this approach.
+
+## When to choose Code-First Approach
+
+When we need to go **fast** in production it is the best approach. Then we can generate our documentation from the API.
+
+Code-First is also good, while the documentation will be generated from the actual Code. Which means, we don't have to keep in **sync** other files.
+
+## Spring Boot 
+
+For our post we will be using [Spring Boot](https://spring.io/projects/spring-boot) together with [springdoc-openapi](https://springdoc.org/).
+
+Springdoc integrates nicely with Spring Boot and will help us generate our documentation.
+
+### Example
+
+#### Getting started
+
+To easily get started we only need to add the Springdoc dependency in our `build.gradle`
+
+```groovy
+implementation 'org.springdoc:springdoc-openapi-ui:1.3.3'
+```
+
+First let us define the path of our documentation. This can be defined easily in the application.yml of our project:
+
+```yaml
+springdoc:
+  api-docs:
+    path: /reflectoring-openapi
+```
+
+This, is where our OpenAPI specification lives, which then springdoc uses to beautifully display our endpoints. For more configuration properties please check the [official documentation](https://springdoc.org/springdoc-properties.html).
+
+#### Define APIs Information
+
+Let us define some information about our API.
+
+```java
+@OpenAPIDefinition(
+        info = @Info(
+                title = "Code-First Approach (reflectoring.io)",
+                description = "" +
+                        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur et rhoncus quam. Aenean quis augue ac eros pulvinar malesuada. " +
+                        "In sagittis elit egestas tincidunt iaculis. " +
+                        "Donec eu lacus vitae nulla varius consectetur a vel quam. Aliquam erat volutpat. Duis eget ullamcorper tellus",
+                contact = @Contact(name = "Reflectoring", url = "https://reflectoring.io", email = "petros.stergioulas94@gmail.com"),
+                license = @License(name = "MIT Licence", url = "https://github.com/thombergs/code-examples/blob/master/LICENSE")),
+        servers = @Server(url = "http://localhost:8080")
+)
+```
+
+Now, start the server and navigate to: http://localhost:8080/swagger-ui/index.html?configUrl=/reflectoring-openapi/swagger-config
```

Comment:
- [ ] How does Swagger come into play here? What's the connection between Swagger and Springdoc? Please explain that somewhere.